packages/fxa-settings/src/pages/Signup/index.tsx

```diff
@@ -0,0 +1,490 @@
+  const onFocusMetricsEvent = 'signup.engage';
+
+  const [age, setAge] = useState<number>(0);
+  const [ageCheckErrorText, setAgeCheckErrorText] = useState('');
+  const [isFocused, setIsFocused] = useState(false);
+  const [newPasswordErrorText, setNewPasswordErrorText] = useState('');
+  const [
+    isAccountSuggestionBannerVisible,
+    setIsAccountSuggestionBannerVisible,
+  ] = useState<boolean>(isPocketClient);
+
+  const { handleSubmit, register, getValues, errors, formState, trigger } =
+    useForm<FormData>({
+      mode: 'onTouched',
+      criteriaMode: 'all',
+      defaultValues: {
+        newPassword: '',
+        confirmPassword: '',
+      },
+    });
+
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  /**
+   * Newsletter data and ftl message resolvers kept here for simplicity,
+   * but could eventually be moved out and imported if the list becomes longer.
+   */
+  const accountsJourneyNewsletterLabel = ftlMsgResolver.getMsg(
+    'signup-newsletter-account-journey',
+    'Get the latest news about Mozilla and Firefox'
+  );
+  const takeActionNewsletterLabel = ftlMsgResolver.getMsg(
+    'signup-newsletter-take-action',
+    'Take action to keep the internet healthy'
+  );
+  const knowledgeIsPowerNewsletterLabel = ftlMsgResolver.getMsg(
+    'signup-newsletter-knowledge-is-power',
+    'Be safer and smarter online'
+  );
+
+  const newsletters: Newsletter[] = [
+    {
+      label: accountsJourneyNewsletterLabel,
+      slug: 'firefox-accounts-journey',
+    },
+    {
+      label: takeActionNewsletterLabel,
+      slug: 'take-action-for-the-internet',
+    },
+    {
+      label: knowledgeIsPowerNewsletterLabel,
+      slug: 'knowledge-is-power',
+    },
+  ];
+
+  /**
+   * List of sync engines is temporarily included here for simplicity and l10n,
+   * but should eventually be moved to a /lib or /model file once functionality
+   * from sync-engine.js is transferred to fxa-settings.
+   */
+
+  const bookmarksEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-bookmarks',
+    'Bookmarks'
+  );
+
+  const historyEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-history',
+    'History'
+  );
+
+  const passwordEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-passwords',
+    'Passwords'
+  );
+
+  const addonsEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-addons',
+    'Add-ons'
+  );
+
+  const openTabsEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-open-tabs',
+    'Open Tabs'
+  );
+
+  const preferencesEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-preferences',
+    'Preferences'
+  );
+
+  const addressesEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-addresses',
+    'Addresses'
+  );
+
+  const creditCardsEngineText = ftlMsgResolver.getMsg(
+    'signup-sync-engine-credit-cards',
+    'Credit Cards'
+  );
+
+  const engines: Engine[] = [
+    {
+      checked: true,
+      id: 'bookmarks',
+      text: bookmarksEngineText,
```

Comment:
Instead of setting each of these to a const, since we only need them once, you could set these to `<FtlMsg>` components, or just set them like:

```
text: ftlMsgResolver.getMsg(
    'signup-sync-engine-bookmarks',
    'Bookmarks'
  );
```

Which feels a little cleaner to me, IMO, and same with newsletters.